### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by eapenzacharias**
Request for code review

### CODE REVIEW

File: app/controllers/post_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class PostController < ApplicationController
```
Comment (dipto0321) [inline app/controllers/post_controller.rb]: Controller class names use CamelCase and have Controller as a suffix. The Controller suffix is always singular. **The name of the resource is usually plural**.

File: app/controllers/post_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  before_action :signedin_user, only: [ :new, :create]
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
```
Comment (dipto0321) [inline app/controllers/post_controller.rb]: There is two way to do this. The convenient way is `current_user.posts.build(post_params)`. Another way is 
```ruby
@post=Post.new(post_params)
@post.user_id = current_user.id
```
Now, it's your choice which one you want to choose.

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Rails.application.routes.draw do
+  get 'post/new'
```
Comment (dipto0321) [inline config/routes.rb]: Can you explain why you need this?

File: config/routes.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  get 'post/create'
```
Comment (dipto0321) [inline config/routes.rb]: Can you explain why you need this?

File: config/routes.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  get 'post/index'
```
Comment (dipto0321) [inline config/routes.rb]: Can you explain why you need this?

File: config/routes.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  get 'static_pages/home'
+  root "post#index"
```
Comment (dipto0321) [inline config/routes.rb]: As I said before. Controller name should be plural. So after changing the controller name, it  should be `posts#index`

File: config/routes.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :sessions, only: [:new, :create, :destroy]
+  match '/signin',  to: 'sessions#new', via: 'get'
```
Comment (dipto0321) [inline config/routes.rb]: Why I need `match` there?

File: config/routes.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  match '/signout', to: 'sessions#destroy', via: 'delete'
```
Comment (dipto0321) [inline config/routes.rb]: Why I need `match` there?

File: config/routes.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  resources :post, only: [:new, :create, :index]
```
Comment (dipto0321) [inline config/routes.rb]: Route names are `snake_case`, and usually match the controller. Most of the time routes are **plural** and use the plural `resources`.

[Singular routes](http://edgeguides.rubyonrails.org/routing.html#singular-resources) are a special case. These use the singular `resource` and a singular resource name. However, they still map to a plural controller by default!

File: config/routes.rb [hunk 10]
```diff
@@ -0,0 +1,14 @@
+Rails.application.routes.draw do
+  get 'post/new'
+  get 'post/create'
+  get 'post/index'
+  get 'static_pages/home'
+  root "post#index"
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :sessions, only: [:new, :create, :destroy]
```
Comment (dipto0321) [inline config/routes.rb]: Another way to do this job with custom naming

```ruby
get '/login', to: 'sessions#new'
post '/login', to: 'sessions#create'
delete '/logout', to: 'sessions#destroy'
```
In that case there no need any resources for the session and also no need any match. As simple as it is.

File: config/routes.rb [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  match '/signin',  to: 'sessions#new', via: 'get'
+  match '/signout', to: 'sessions#destroy', via: 'delete'
+  resources :post, only: [:new, :create, :index]
+  match '/post/new',  to: 'post#new', via: 'post'
```
Comment (dipto0321) [inline config/routes.rb]: Why need this?

File: config/routes.rb [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+  match '/post/create',  to: 'post#create', via: 'post'
```
Comment (dipto0321) [inline config/routes.rb]: Why need this?

File: app/views/post/new.html.erb [hunk 13]
```diff
@@ -0,0 +1,16 @@
+<%= provide(:title, "Create post") %>
+<h1>Create Posts</h1>
+
+
+<%= form_for(:post, url: {action: "create"}) do |f| %>
+  <%= f.hidden_field :user_id, value: current_user.id %>
```
Comment (dipto0321) [inline app/views/post/new.html.erb]: Why `user_id` need to pass as params? The way I mentioned before for posts controller will solve this. This is not the most convenient way to do that job.

File: app/controllers/post_controller.rb [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  def signedin_user
+    unless signed_in?
+      redirect_to signin_path
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :content, :user_id)
```
Comment (dipto0321) [inline app/controllers/post_controller.rb]: No need to pass :user_id as params.

File: config/routes.rb (hunk 9 above)
Comment (dipto0321) [reply config/routes.rb]: View file names, by default, match the controller and action that they are tied to.

Views go in `app/views/#{resource_name}/#{action_name}.html.erb`.

In an example, For posts resource, and index actions. Views go in `app/views/posts/index.html.erb`.

File: app/controllers/post_controller.rb (hunk 2 above)
Comment (lilleonz) [reply app/controllers/post_controller.rb]: chose the build option

File: config/routes.rb [hunk 15]
```diff
@@ -0,0 +1,12 @@
+Rails.application.routes.draw do
+  root "posts#index"
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :user, only: [:new, :create]
```
Comment (dipto0321) [inline config/routes.rb]: This should be `:users`. Plural.

File: config/routes.rb [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+  resources :sessions, only: [:new, :create, :destroy]
+  get '/signup', to: 'user#new'
```
Comment (dipto0321) [inline config/routes.rb]: Either use this

```ruby
resources :users, only: [:new, :create] 
```
Or 
```ruby
get '/signup', to: 'user#new'
post '/signup', to: 'user#create'
```
So remove one.

Comment (dipto0321) [reply config/routes.rb]: Another thing. Again like I said for `posts` controller. User should be Users also in the controller. `UsersController` like this. So the method should be `users#new`. Also, fix this.

File: config/routes.rb [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+  post '/signup', to: 'user#create'
```
Comment (dipto0321) [inline config/routes.rb]: Already mentioned ☝️

File: config/routes.rb [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+
+  match '/signin',  to: 'sessions#new', via: 'get'
```
Comment (dipto0321) [inline config/routes.rb]: The same thing as users sign up. If there is a resource then no need this way to create a route. So remove one of them. Either `resources :sessions` part or those match.

Comment (lilleonz) [reply config/routes.rb]: Dipto Please see comments in the routes.rb about the sessions issues
Leon

File: app/controllers/posts_controller.rb [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @posts = Post.all
+  end
+
+  def signedin_user
```
Comment (dipto0321) [inline app/controllers/posts_controller.rb]: Move this method after `private`. This should be a private method.

File: config/routes.rb [hunk 20]
```diff
@@ -0,0 +1,51 @@
+Rails.application.routes.draw do
+  root 'posts#index'
+  # resources :users, only: [:new, :create]
+  get '/signup', to: 'users#new'
+  post '/signup', to: 'users#create'
+
+  resources :sessions, only: [:new, :create, :destroy]
```
Comment (dipto0321) [inline config/routes.rb]: I read your comments and the cool part is I already gave you the solution. Somehow you missed it. You can remove that `resources :sessions` and you have to add `post '/signin', to: 'sessions#create'`. You can see ☝️ which I clearly mentioned. I already tested it and it works in my case without any error.

**Comment by lilleonz** [comment]
awesome! thank you for the tips. will take care of it and merge! :D

**Pull request merged**